solidity/test/rewards/TestECDSARewards.js

```diff
@@ -0,0 +1,232 @@
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("interval allocation", async () => {
+    expectedKEEPAllocation = [
+      7128000.0, 13685760.0, 15738624.0, 16997713.92, 18697485.31, 15892862.52,
+      13508933.14, 11482593.17, 9760204.19, 8296173.56, 7051747.53, 5993985.4,
+      5094887.59, 4330654.45, 3681056.28, 3128897.84, 2659563.16, 2260628.69,
+      1921534.39, 1633304.23, 1388308.59, 1180062.31, 1003052.96, 852595.02
+    ]
+
+    it("should equal to expected allocations when 5 keeps were created per interval", async () => {
+      await verifyIntervalAllocations(5)
+    })
+
+    it("should equal to expected allocations when 1 keep was created per interval", async () => {
+      await verifyIntervalAllocations(1)
+    })
+  })
+
+  async function verifyIntervalAllocations(keepToCreatePerInterval) {
+    let keepCreationTimestamp = firstKeepCreationTimestamp
+
+    for (let i = 0; i < 24; i++) {
+      await createKeeps(keepToCreatePerInterval, keepCreationTimestamp)
+
+      keepCreationTimestamp = await timeJumpToEndOfInterval(i)
+
+      await rewardsContract.allocateRewards(i)
+
+      const actualBalance = (await rewardsContract.getAllocatedRewards(i)).div(
+        tokenDecimalMultiplier
+      )
+
+      expect(actualBalance).to.gte.BN(expectedKEEPAllocation - precision)
+      expect(actualBalance).to.lte.BN(expectedKEEPAllocation + precision)
+    }
+  }
+
+  describe("rewards distribution", async () => {
+    it("should not be possible when a keep is not closed", async () => {
+      await createKeeps(1, firstKeepCreationTimestamp)
+
+      const keepAddress = await keepFactory.getKeepAtIndex(0)
+
+      const isEligible = await rewardsContract.eligibleForReward(keepAddress)
+      expect(isEligible).to.be.false
+
+      await timeJumpToEndOfInterval(0)
+      await expectRevert(
+        rewardsContract.receiveReward(keepAddress),
+        "Keep is not closed"
+      )
+    })
+
+    it("should not be possible when a keep is terminated", async () => {
+      await createKeeps(1, firstKeepCreationTimestamp)
+
+      const keepAddress = await keepFactory.getKeepAtIndex(0)
+      const keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsTerminated()
+
+      const isEligible = await rewardsContract.eligibleForReward(keepAddress)
+      expect(isEligible).to.be.false
+
+      await timeJumpToEndOfInterval(0)
+      await expectRevert(
+        rewardsContract.receiveReward(keepAddress),
+        "Keep is not closed"
+      )
+    })
+
+    it("should not count terminated groups when distributing rewards", async () => {
+      await createKeeps(2, firstKeepCreationTimestamp)
+      // reward for the first interval: 7128000 KEEP
+      // keeps created: 2, but the min is 4 => 7128000 / 4 = 1782000 KEEP per keep
+      // member receives: 1782000 / 3 = 594000 (3 signers per keep)
+      const expectedBeneficiaryBalance = new BN(594000)
```

Comment:
Can we move this calculation to be next to assertion? The value expected is `594000` and not `2*594000` because the keep was terminated and it's happening after the expected value is declared. It's a little confusing.

Reply:
Done.